src/main/java/io/odpf/firehose/sink/prometheus/builder/TimeSeriesBuilder.java

```diff
@@ -0,0 +1,182 @@
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.Duration;
+
+import cortexpb.Cortex;
+
+import java.util.Properties;
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import static io.odpf.firehose.sink.prometheus.PromSinkConstants.*;
+
+/**
+ * Builder for Cortex TimeSeries.
+ *
+ * The buildTimeSeries(Message message, int partition) method should be synchronised if called from multiple threads
+ */
+public class TimeSeriesBuilder {
+
+    private Cortex.TimeSeries.Builder cortexTimeSeriesBuilder = Cortex.TimeSeries.newBuilder();
+    private Cortex.LabelPair.Builder cortexLabelBuilder = Cortex.LabelPair.newBuilder();
+    private Cortex.Sample.Builder cortexSampleBuilder = Cortex.Sample.newBuilder();
+
+    private Properties metricNameProtoIndexMapping;
+    private Properties labelNameProtoIndexMapping;
+    private Integer timestampIndex;
+    private Boolean isEventTimestampEnabled;
+
+    /**
+     * Instantiates a new cortex time series builder.
+     *
+     * @param config    the prometheus sink config
+     */
+    public TimeSeriesBuilder(PromSinkConfig config) {
+        metricNameProtoIndexMapping = config.getSinkPromMetricNameProtoIndexMapping();
+        labelNameProtoIndexMapping = config.getSinkPromLabelNameProtoIndexMapping();
+        timestampIndex = config.getSinkPromProtoEventTimestampIndex();
+        isEventTimestampEnabled = config.isEventTimestampEnabled();
+    }
+
+    /**
+     * build list of sorted cortex time series object.
+     *
+     * @param message                           the protobuf message
+     * @param partition                         the kafka partition where the message is consumed
+     * @return                                  list of sorted cortex time series object
+     * @throws InvalidProtocolBufferException   the exception on invalid protobuf
+     */
+    public List<Cortex.TimeSeries> buildTimeSeries(Message message, int partition) throws InvalidProtocolBufferException {
+        Map<String, Object> labelPair = getLabelMessage(message, labelNameProtoIndexMapping, partition);
+        List<Map<String, Object>> metricList = getMetricMessage(message, metricNameProtoIndexMapping);
+        List<Cortex.TimeSeries> timeSeriesList = new ArrayList<>();
+        Long metricTimestamp = getMetricTimestamp(message);
+        for (Map<String, Object> metricName : metricList) {
+            buildMetric((String) metricName.get(METRIC_NAME));
+            for (Map.Entry<String, Object> entry : labelPair.entrySet()) {
+                buildLabels(entry.getKey(), entry.getValue());
+            }
+            buildSample(metricTimestamp, Double.parseDouble(metricName.get(METRIC_VALUE).toString()));
+            timeSeriesList.add(cortexTimeSeriesBuilder.build());
+            cortexTimeSeriesBuilder.clear();
+        }
+        return timeSeriesList;
+    }
```

Comment:
I was thinking about this again.. let's create the builders(timeseries,lable, sample) inside this method and not put as a instance variable... 
then pass them to buildMetric, buildLabels and buildSample methods..

this is not very readable..
here you are calling cortexTimeSeriesBuilder.build().. but doesn't show how it was updated.

Reply:
yes, i think so. but rather than that, i was thinking on put the builders on respective method.

```
private Cortex.LabelPair buildMetric(String metricName) {
        return Cortex.LabelPair.newBuilder()
                .setName(PROMETHEUS_LABEL_FOR_METRIC_NAME)
                .setValue(metricName)
                .build();
    }
```

and with this i think we dont need to call clear() method. (except on cortexTimeseriesBuilder).
hdyt?

Reply:
It's alright to create one builder per buildTimeSeries() call and then call clear it before setting a new one. 
I am just thinking would that be too many newBuilder() calls if we go the other way. 
```
public List<Cortex.TimeSeries> buildTimeSeries(Message message, int partition) throws InvalidProtocolBufferException {
    Cortex.TimeSeries.Builder cortexTimeSeriesBuilder = Cortex.TimeSeries.newBuilder();
    Cortex.LabelPair.Builder cortexLabelBuilder = Cortex.LabelPair.newBuilder();
    Cortex.Sample.Builder cortexSampleBuilder = Cortex.Sample.newBuilder();
    ...
    ...
    buildMetric(metricName, cortexTimeSeriesBuilder, cortexLabelBuilder);
}
```